### PR TITLE
feat: add metadata

### PR DESCRIPTION
**Pull request by mfiedorowicz**
This pull request adds support for attaching custom metadata to ingestion requests and entities in the Diode Go SDK. Users can now provide additional context or tracking information at both the entity and request level, and this metadata is preserved across gRPC, dry run, and OTLP export flows. The documentation and test suite have been updated to demonstrate and verify these new capabilities.

**Metadata support and API changes:**

* Added a new `WithIngestMetadata` option and supporting types to allow passing metadata (as key-value pairs) with `Ingest` and `IngestProto` calls in both `GRPCClient` and `DryRunClient`. This metadata is attached to the `IngestRequest` and serialized as a protobuf struct. [[1]](diffhunk://#diff-754428b2d0c15530c0f78da3f3d46a161120fd38968a8ed8ae9c153a005e9287L205-R225) [[2]](diffhunk://#diff-754428b2d0c15530c0f78da3f3d46a161120fd38968a8ed8ae9c153a005e9287L520-R548) [[3]](diffhunk://#diff-754428b2d0c15530c0f78da3f3d46a161120fd38968a8ed8ae9c153a005e9287R561-R569) [[4]](diffhunk://#diff-e9a55596ddc9960faef1a74c0a5ca4afd26485ada827b62ca68196a60b71721aL59-R85)
* Updated the client interfaces and implementations to accept variadic options for ingest operations, ensuring backward compatibility. [[1]](diffhunk://#diff-754428b2d0c15530c0f78da3f3d46a161120fd38968a8ed8ae9c153a005e9287L205-R225) [[2]](diffhunk://#diff-e9a55596ddc9960faef1a74c0a5ca4afd26485ada827b62ca68196a60b71721aL59-R85)

**Documentation improvements:**

* Expanded the `README.md` with detailed usage examples for entity-level and request-level metadata, including how metadata is handled in dry run and OTLP export scenarios. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R122-R288) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R350-R379) [[3]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R430-R470)

**Testing enhancements:**

* Added comprehensive tests to verify metadata handling in `Ingest`, `IngestProto`, and dry run flows, including tests for various metadata types (primitives, arrays, nested objects, and edge cases). Ensured that metadata is correctly serialized and included in requests. [[1]](diffhunk://#diff-5b0704091b945f4382d427b699d48076b98041a69fb48bb5e1563756bcdc3d3bR1206-R1502) [[2]](diffhunk://#diff-955649906a2a759d331a53dba7f1c3f15eeee11ee6b25ba639e2ccf512d29f23R165-R209)

**Internal codebase updates:**

* Imported necessary protobuf packages for struct and JSON serialization to support the new metadata functionality. [[1]](diffhunk://#diff-754428b2d0c15530c0f78da3f3d46a161120fd38968a8ed8ae9c153a005e9287R28-R29) [[2]](diffhunk://#diff-e9a55596ddc9960faef1a74c0a5ca4afd26485ada827b62ca68196a60b71721aR14) [[3]](diffhunk://#diff-5b0704091b945f4382d427b699d48076b98041a69fb48bb5e1563756bcdc3d3bR24)

These changes make the SDK more flexible and extensible for tracking, auditing, and enriching ingested data with custom attributes.

### CODE REVIEW

File: diode/dryrun_test.go [hunk 1]
```diff
@@ -162,3 +162,48 @@ func TestDryRunClientSDKVersionCaching(t *testing.T) {
...
+	require.NoError(t, err)
+
+	require.NoError(t, drc.Close())
+
+	// Verify two files were created
+	entries, err := os.ReadDir(dir)
+	require.NoError(t, err)
+	require.Len(t, entries, 2)
+
+	// Load the first file (with metadata) and verify metadata is present
```
Comment (jajeffries) [inline diode/dryrun_test.go:202]: This test doesn't verify the metadata is present in what's returned from `LoadDryRunEntities`. Could we add an asserrtion to verify it's there?

**Review by jajeffries — APPROVED** [review]
Overall looks good, but one minor suggestion about the dry run client

File: diode/dryrun_test.go (hunk 1 above)
Comment (mfiedorowicz) [reply diode/dryrun_test.go:202]: `LoadDryRunEntities()` returns entities only, not the full request

Comment (mfiedorowicz) [reply diode/dryrun_test.go:202]: Added explicit metadata assertions on the ingest request loaded from dryrun output below

**Review by leoparente — APPROVED** [review]

**Review by jajeffries — APPROVED** [review]
